proxy/http/HttpSM.cc

```diff
@@ -6560,6 +6559,12 @@ HttpSM::kill_this()
   reentrancy_count--;
   ink_release_assert(reentrancy_count == 0);
 
+  // Delay the close of the user agent session, so the close session
+  // occurs after the close transaction
+  if (ua_session != NULL)
```

Comment:
Ah, in the KA case, ua_session->release() will be called in tunnel_handler_ua() and ua_session will be set to null before kill_this is called.
I'll go ahead and explicitly test that case too.